app/components/database-version-selector/component.js

```diff
@@ -0,0 +1,18 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  selectedDatabaseImage: Ember.computed('visibleDatabaseImages', function() {
+    let dbImage = this.get('visibleDatabaseImages').find((image) => image.get('default'));
+    this.get('dbImage');
+    this.set('dbImage', dbImage);
+    return dbImage;
+  }),
+
+  visibleDatabaseImages: Ember.computed('type', function() {
+    return this.get('databaseImages').filter((image) => image.get('type') == this.get('type'))
+  }),
+
+  selectImage: function () {
+    this.sendAction('imageSelected', this.get('dbImage'));
+  }.observes('dbImage')
```

Comment:
I think rather than observing the mutated dbImage, you can just pass in "selectImage" as the `update` attribute on the component instantiation.  That way, you can remove the `observes` which is kind of an anti-pattern. 

```hbs
// template.hbs
{{ ultimate-select
  items=visibleDatabaseImages
  name="databaseImage"
  itemKey="id"
  itemValue="description"
  update="selectImage"
  selected=selectedDatabaseImage }} 
```

```javascript
// component.js
selectImage(dbImage) {
  this.set('dbImage', dbImage);
  this.sendAction('imageSelected', this.get('dbImage'));
}
```